src/MySqlConnector/MySqlClient/Results/ResultSet.cs

```diff
@@ -46,9 +48,31 @@ public ResultSet(MySqlDataReader dataReader)
 					if (State == ResultSetState.NoMoreData)
 						break;
 				}
-				else if (firstByte == 0xFB)
+				else if (firstByte == LocalInfilePayload.Signature)
 				{
-					throw new NotSupportedException("Don't support LOCAL_INFILE_Request");
+					try
+					{
+						var localInfile = LocalInfilePayload.Create(payload);
+						using (var stream = localInfile.FileName.StartsWith(MySqlBulkLoader.StreamPrefix, StringComparison.Ordinal) ?
+							MySqlBulkLoader.GetAndRemoveStream(localInfile.FileName) :
+							File.OpenRead(localInfile.FileName))
+						{
+							byte[] readBuffer = new byte[65536];
```

Comment:
65536 seems like a logical size though.  Not too much RAM but should be enough to keep the socket buffer busy.